dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -1,5 +1,322 @@
+            {{ current_timestamp() }} AS dbt_valid_from {%- if not loop.last -%},{%- endif -%}
+          {% elif column.name == 'dbt_change_type' %}
+            'delete' AS dbt_change_type {%- if not loop.last -%},{%- endif -%}
+          {% elif column.name == 'dbt_valid_to' %}
+            CAST('9999-01-01' as timestamp) AS dbt_valid_to {%- if not loop.last -%},{%- endif -%}
+          {% elif column.name == 'is_current_record' %}
+            True AS is_current_record {%- if not loop.last -%},{%- endif -%}
+          {% else %}
+            snapshotted_data.{{ column.name }} {%- if not loop.last -%},{%- endif -%}
+          {% endif %}
+        {% endfor %}
+        FROM snapshotted_data
+        LEFT JOIN source_data
+               ON snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
+        WHERE source_data.dbt_unique_key IS NULL
+    )
+    SELECT * FROM upserts
+    UNION ALL
+    SELECT * FROM deletes;
+    {% else %}
+    SELECT * FROM upserts;
+    {% endif %}
+
+{%- endmacro %}
+
+{%
+  Create a new temporary table to hold the records to be upserted or deleted
+%}
+
+{% macro athena__build_snapshot_staging_table(strategy, sql, target_relation) %}
+    {%- set tmp_identifier = target_relation.identifier ~ '__dbt_tmp' -%}
+
+    {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
+                                                  schema=target_relation.schema,
+                                                  database=none,
+                                                  type='table') -%}
+
+    {% do adapter.drop_relation(tmp_relation) %}
+
+    {%- set select = snapshot_staging_table(strategy, sql, target_relation) -%}
+
+    {% call statement('build_snapshot_staging_relation') %}
+        {{ create_table_as(False, tmp_relation, select) }}
+    {% endcall %}
+
+    {% do return(tmp_relation) %}
+{% endmacro %}
+
+{#
+    Add new columns to the table if applicable
+#}
+
+{% macro create_columns(relation, columns) -%}
+  {{ adapter.dispatch('create_columns', 'dbt')(relation, columns) }}
+{% endmacro %}
+
+{% macro default__create_columns(relation, columns) -%}
+  {%- for column in columns -%}
+    {% if column.data_type|lower == 'boolean' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} BOOLEAN);
+    {%- endset -%}
+    {% elif column.data_type|lower == 'character varying(256)' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} VARCHAR);
+    {%- endset -%}
+    {% elif column.data_type|lower == 'integer' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} INT);
+    {%- endset -%}
+    {% elif column.data_type|lower == 'float' %}
+    {% set query -%}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} FLOAT);
+    {%- endset -%}
+    {% else %}
+      ALTER TABLE {{ relation }} ADD COLUMNS ({{ column.name }} {{ column.data_type }});
+    {% endif %}
```

Comment:
plese consider to re-write this and make use of https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/utils/ddl_data_type.sql#L3

Reply:
We could consider to do the suggested refactor, when we introduce iceberg for snapshots.

Reply:
I was trying to get this to work but for some reason the ddl_data_type macro returned nothing when I tried

```
dbt -d run-operation ddl_data_type --args '{col_type: integer}'
```

Will keep this in mind for improvement when introducing ICEBERG